### ISSUE TITLE
External package reference

### ISSUE THREAD

**Issue by Prn-Ice** [myConsciousness/freezer#41]
# 1. Problem to solve

First, thanks for an awesome package yet again.
Say I have a model or set of models in an external package, I'd like to be able to add the external package import path as a reference.

# 2. Proposal

- To import a `user` model from package `user_repository`.
- In references add `package:user_repository`.
- Generated code uses import url and skips generation of `user`.

# 3. More information

I would be interested in working on a PR for this with some direction.

**Comment by myConsciousness** [comment]
Hi @Prn-Ice , thanks for your feedback :)

You can create a PR for this issue, and I imagine it would be like below:

```json
{
   "models":{
      "something":{
         "user":{
            "name":""
         }
      }
   },
   "references":{
      "user":"package:user_repository/user_repository.dart"
   }
}
```

And generated object will be:

```dart
// GENERATED CODE - DO NOT MODIFY BY HAND

import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:user_repository/user.dart';

part 'something.freezed.dart';
part 'something.g.dart';

// **************************************************************************
// FreezerGenerator
// **************************************************************************

@freezed
class Something with _$Something {
  const factory Something({
    User? user,
  }) = _Something;

  /// Returns [Something] based on [json].
  factory Something.fromJson(Map<String, Object?> json) => _$ShopFromJson(json);
}
```

It would be a great feature :)

**Comment by Prn-Ice** [comment]
Will do

**Comment by Prn-Ice** [comment]
Any pointers to a good starting point?

**Comment by myConsciousness** [comment]
Merged #42
